runtime/pkg/formatter/format_measure_value.go

```diff
@@ -0,0 +1,86 @@
+package formatter
+
+import (
+	"fmt"
+	"strings"
+
+	// "d3-format" equivalent needed
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"go.uber.org/zap"
+)
+
+// MeasureValueFormatter formats measure values
+// The code is based on the logic implemented on the frontend
+// Type and file names are kept mostly the same for consistency and ease of updating
+type MeasureValueFormatter struct {
+	formatter
+	logger *zap.Logger
+}
+
+func (f *MeasureValueFormatter) Format(value any) string {
+	str, err := f.stringFormat(value)
+	if err == nil {
+		return str
+	}
+	f.logger.Warn("failed to format value, returning non-formatted value", zap.Any("value", value), zap.Error(err))
```

Comment:
We should avoid logging from this util library. Better to return an error and then log the error in the caller if desired.